frontend/src/APIClients/AuthAPIClient.ts

```diff
@@ -1,80 +1,134 @@
-import baseAPIClient from "./BaseAPIClient";
 import AUTHENTICATED_USER_KEY from "../constants/AuthConstants";
 import { AuthenticatedUser } from "../contexts/AuthContext";
-import {
-  getLocalStorageObjProperty,
-  setLocalStorageObjProperty,
-} from "../utils/LocalStorageUtils";
+// rest {
+// import baseAPIClient from "./BaseAPIClient";
+// import {
+//   getLocalStorageObjProperty,
+//   setLocalStorageObjProperty,
+// } from "../utils/LocalStorageUtils";
+// } rest
+// graphql {
+import { setLocalStorageObjProperty } from "../utils/LocalStorageUtils";
+// } graphql
 
+// graphql {
 const login = async (
   email: string,
   password: string,
-): Promise<AuthenticatedUser> => {
-  try {
-    const { data } = await baseAPIClient.post(
-      "/auth/login",
-      { email, password },
-      { withCredentials: true },
-    );
-    localStorage.setItem(AUTHENTICATED_USER_KEY, JSON.stringify(data));
-    return data;
-  } catch (error) {
-    return null;
+  loginFunction: Function,
```

Comment:
Minor thing but let's type the function more specifically, you can import the missing types (e.g. `MutationFunctionOptions` from `@apollo/client`.

```ts
loginFunction: (
    options?:
      | MutationFunctionOptions<
          { login: AuthenticatedUser },
          OperationVariables
        >
      | undefined,
  ) => Promise<
    FetchResult<
      { login: AuthenticatedUser },
      Record<string, any>,
      Record<string, any>
    >
  >
```

Reply:
thanks, made the changes 👍 and will wait till 42 is merged first